### DISCUSSION TITLE
[HELP] Set Minimum width for the Shell

### DISCUSSION THREAD

**Discussion by TWEESTY** [dotnet/maui#13066] [Q&A] [answered]
Hello everyone,

I try to set a the minimum width for my application but it doesn't work.
First, I try to set the property "MinimumWidthRequest" for my "Shell" element.

Then I discovered this discussion about setting a fixed size at the startup : [https://github.com/dotnet/maui/discussions/2370](https://github.com/dotnet/maui/discussions/2370). So I understood, it will no be as simple to set the minimum.

For UWP, I found how to do what I want : [https://github.com/microsoft/microsoft-ui-xaml/issues/2945](https://github.com/microsoft/microsoft-ui-xaml/issues/2945). But I cannot succeed to implement this in my "App.xaml.cs" in my directory Windows platform.

Someone can help me ? :)

Thanks

**Comment by twojnarowski** [comment, marked as answer]
I am also looking for an answer for this problem. I looked at https://github.com/dotnet/maui/discussions/2370 and if you reach the point when you have a variable with `AppWindow`, then you can hook up to the `AppWindow.SizeChanged += NativeWindow_SizeChanged;` event and in there protect from being too small:
```cs
private static void NativeWindow_SizeChanged(object sender, Microsoft.UI.Xaml.WindowSizeChangedEventArgs args)
{
    if (args.Size.Width * 1.05 < 896)
    {
        int height = (int)(args.Size.Height * DeviceDisplay.MainDisplayInfo.Density);

        AppWindow.Resize(new(896, height));
    }

    if (args.Size.Height * 1.05 < 720)
    {
        int width = (int)(args.Size.Width * DeviceDisplay.MainDisplayInfo.Density);

        AppWindow.Resize(new(width, 720));
    }

    args.Handled = true;
}
```

It doesn't look as it should for a desktop app, it can be made smaller and just springs back into the "minimum size", but at least its something...
And that 1.05 is a random magic number I found, because for some reason (even though my DPI density is 1, the args report the size to be just 1 or 2% smaller. So without multiplying by 1.05, the whole thing goes into a loop constantly making it smaller and smaller by itself.
